apps/src/code-studio/components/report_abuse_form.jsx

```diff
@@ -67,7 +67,13 @@ export default class ReportAbuseForm extends React.Component {
    * @returns {string} Channel id, or undefined if we can't get one.
    */
   getChannelId() {
-    const match = /.*\/projects\/[^\/]+\/([^\/]+)/.exec(this.props.abuseUrl);
+    const abuseUrl = this.props.abuseUrl;
+    let match;
+    if (abuseUrl.indexOf('codeprojects') >= 0) {
+      match = /.*codeprojects.*[^\/]+\/([^\/]+)/.exec(abuseUrl);
+    } else {
+      match = /.*\/projects\/[^\/]+\/([^\/]+)/.exec(abuseUrl);
+    }
```

Comment:
Presumably we could instead do something like:
`.*\/(?:code)?projects\/[^\/]+\/([^\/]+)`

(i.e. we add a non-capturing group of `code` in front of projects

Reply:
I don't think this would work for localhost.codeproject.org:3000/ because there is no `/` before code

Reply:
I see.

It feels a little strange to do some of our string checking using indexOf and some using a regex. It certainly should be possible to construct a regex that covers both cases, but perhaps such a regex would become too complex.

Either way, I wonder if it's worth (a) extracting this as a pure method, i.e. something like `function getChannelIdFromUrl(url)` and then (b) write a quick unit test that covers both projects and codeprojects urls in different environments.